docs/architecture/component_templates.md

```diff
@@ -0,0 +1,134 @@
+# Account component templates
+
+An account component template provides a general description of an account component. It encapsulates all the information needed to initialize and manage the component.
+
+Specifically, a template specifies a component's **metadata** and its **code**.
+
+Once defined, a component template can be instantiated as account components, which can then be grouped to form the account's `Code`.
+
+## Component code
+
+The component template’s code defines a library of functions that operate on the specified storage layout.
+
+## Component metadata
+
+The component metadata describes the account component entirely: its name, description, version, and storage layout.
+
+The storage layout can optionally define specific initial values for each of the slots. Alternatively, placeholders can be utilized to identify values that should be provided at the moment of instantiation.
+
+### TOML specification
+
+The component metadata can be defined using TOML. Below is an example specification:
+
+```toml
+name = "Fungible Faucet"
+description = "This component displays valid values for component templates"
+version = "1.0.0"
+supported-types = ["FungibleFaucet"]
+
+[[storage]]
+name = "token_metadata"
+description = "Contains metadata about the token associated to the faucet account. The metadata is formed by three fields: max supply, the token symbol and the asset's decimals"
+slot = 0
+value = [
+    { type = "felt", name = "max_supply", description = "Maximum supply of the token in base units" },
+    { type = "tokensymbol", value = "TST" },
```

Comment:
Not related to the docs, but should this be `token_symbol` instead?
More generally, do we have a convention for the template types (snake case, camel case, ...)? For the rpo key we're using snake case.

Reply:
I agree that we should probably specify the naming convention (even if we are not enforcing it), and if we do so, I'd probably go with snake case.

Reply:
Agreed! Opened https://github.com/0xPolygonMiden/miden-base/pull/1215

Reply:
[TOML does not allow inline tables to have newlines](https://toml.io/en/v1.0.0#inline-table), so no very simple way to do this other than using full tables which does not look too nice either.